### PR TITLE
#32 ORCID validation case

### LINKED ISSUE #32: Checking 16-digit ORCID identifier
@omaus, @kMutagene 

the Orcid identifier must correspond to the following scheme: xxxx-xxxx-xxxx-xxxx. Otherwise, no valid record can be created in Invenio. This must be checked, otherwise Archigator will create a "broken" Invenio record. 

The current example is the ARCInvenioTesting repository, where "orcid1", "orcid2" etc. are entered as identifiers in the isa file. The creation of an Invenio record fails here.

### PR DESCRIPTION
**Pull request by omaus**
This PR

- adds a Regex pattern that every ORCID entry must match to succeed the validation
- fixes #32

### CODE REVIEW

**Review by kMutagene — CHANGES_REQUESTED** [review]
Also, there are no tests for this (in this repo).

these are valid orcids:

https://orcid.org/0000-0001-5109-3700
https://orcid.org/0000-0002-1694-233X

File: src/ArcValidation/Validate/Critical/ValidateCvBase.fs [hunk 1]
```diff
@@ -41,7 +41,13 @@ module CvBase =
 
         /// Validates a person's ORCID.
         let orcid<'T when 'T :> CvContainer> (personCvContainer : 'T) = 
-            property "<  ORCID>" personCvContainer
+            let orcidProperty = CvContainer.tryGetPropertyStringValue "<  ORCID>" personCvContainer
+            let rgxPat = System.Text.RegularExpressions.Regex("^\d{4}-\d{4}-\d{4}-\d{4}$")
```
[the remark this answers is not in the document]
Comment (omaus) [reply src/ArcValidation/Validate/Critical/ValidateCvBase.fs:45]: Yes, this was also already mentioned in #32: https://github.com/nfdi4plants/arc-validate/issues/32#issuecomment-1641081529

**Review by kMutagene — CHANGES_REQUESTED** [review]
Please add unit tests for the newly added functions (checksum, checkValid)

**Comment by kMutagene** [comment]
Also correct me if I am wrong, but i think your checks do not cover the allowed number ranges

**Review by kMutagene — APPROVED** [review]

**Pull request merged**